microcosm_pubsub/chain/context_decorators.py

```diff
@@ -0,0 +1,101 @@
+from functools import wraps
+from marshmallow import ValidationError
```

Comment:
Technically: `functools` and `inspect` are built-ins, `marshmallow` is a library